src/main/webapp/resources/js/pages/projects/share/index.js

```diff
@@ -14,23 +32,77 @@ import store from "./store";
  * @constructor
  */
 function ShareApp() {
+  const [step, setStep] = React.useState(0);
+  const [prevDisabled, setPrevDisabled] = React.useState(true);
+  const [nextDisabled, setNextDisabled] = React.useState(true);
+  const [error, setError] = React.useState(undefined);
+  const [finished, setFinished] = React.useState(false);
+
   /*
   Create redirect href to project samples page.
   */
   const [redirect] = React.useState(
     () => window.location.href.match(/(.*)\/share/)[1]
   );
 
-  const { originalSamples, currentProject } = useSelector(
-    (state) => state.shareReducer
+  const {
+    samples = [],
+    currentProject,
+    locked,
+    targetProject,
+    remove,
+    metadataRestrictions,
+  } = useSelector((state) => state.shareReducer);
+
+  const [shareSamplesWithProject] = useShareSamplesWithProjectMutation();
+
+  const { data: existingIds = [] } = useGetSampleIdsForProjectQuery(
+    targetProject?.identifier,
+    {
+      skip: !targetProject?.identifier,
+    }
   );
 
+  const { data: projects, isLoading: projectsLoading } =
+    useGetPotentialProjectsToShareToQuery(currentProject, {
+      skip: !currentProject,
+    });
+
+  const filtered = samples.filter((sample) => !existingIds.includes(sample.id));
+
+  const steps = [
+    {
+      title: i18n("ShareLayout.project"),
+      component: <ShareProject projects={projects} />,
+    },
+    {
+      title: i18n("ShareLayout.samples"),
+      component: <ShareSamples samples={filtered} />,
+    },
+    { title: i18n("ShareLayout.restrictions"), component: <ShareMetadata /> },
```

Comment:
Can we update this so this step is only displayed if there are actual metadata fields? Right now if it has none the step still gets displayed

Reply:
I will think about this and get back to you this morning